src/rucksack-3-mkdir.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/src/main.c b/src/main.c
+index 24583e8..006a739 100644
+--- a/src/main.c
++++ b/src/main.c
+@@ -1360,7 +1360,7 @@ static int rucksack_mkdirp(const char *path) {
+     if (!err && S_ISDIR(st.st_mode))
+         return 0;
+ 
+-    err = mkdir(path, 0777);
++    err = mkdir(path);
```

Comment:
How about this upstream commit?

https://github.com/andrewrk/rucksack/compare/bf4401b71c2849eab97477eef220e7e9dc516251...master

Once we resolve the rest of the issues with this PR I'll cut a new bugfix release.